src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,39 @@
+package duke.command;
+
+import duke.exceptions.DukeException;
+import duke.io.UserOutputHandler;
+import duke.tasks.TaskList;
+
+/**
+ * Represents a user command.
+ *
+ * @author kevin9foong
+ */
+public abstract class Command {
+    private final String userInputBody;
+
+    protected Command(String userInputBody) {
+        this.userInputBody = userInputBody;
+    }
+
+    /**
+     * Handles this command.
+     *
+     * @param userOutputHandler handles outputting messages to the output destination.
+     * @param taskList          handles task operations including adding, deleting, marking as done and retrieval.
+     * @throws DukeException thrown when a failure occurs within the Duke chat bot.
+     */
+    public abstract void execute(UserOutputHandler userOutputHandler, TaskList taskList)
+            throws DukeException;
```

Comment:
I feel like there's no need to put the throws part in a new line; as I've counted, putting the entire method signature in one line won't make the line longer than 120 characters :D